frontend/packages/dev-console/src/components/import/SamplesCatalog.tsx

```diff
@@ -19,16 +26,31 @@ const SampleCatalog: React.FC<SampleCatalogProps> = ({ match }) => {
       <NamespacedPage variant={NamespacedPageVariants.light} hideApplications>
         {namespace ? (
           <CatalogServiceProvider namespace={namespace} catalogId="samples-catalog">
-            {(service) => (
-              <CatalogController
-                {...service}
-                hideSidebar
-                title={t('devconsole~Samples')}
-                description={t(
-                  'devconsole~Get Started using applications by choosing a code sample.',
-                )}
-              />
-            )}
+            {(service) => {
+              let filteredLists: CatalogItem[];
+              if (sampleType) {
+                filteredLists = service.items.filter((item) => {
+                  return (
+                    item?.typeLabel === labelFilter?.value ||
+                    item?.data?.metadata?.labels[labelFilter?.label] === labelFilter?.value
+                  );
+                });
+              } else {
+                filteredLists = service.items;
+              }
+              const catalogItems = { ...service, items: filteredLists };
```

Comment:
We can talk about the alternative here, but for now I would recommend to focus on `sampleType` filter. A `labelFilter` might have more powers, but we shouldn't mix both.

```suggestion
              let filteredItems = service.items;
              if (sampleType) {
                filteredItems = filteredItems.filter((item) => item.typeLabel === sampleType);
              }
              const catalogItems = { ...service, items: filteredItems };
```

Maybe I miss something, but as long as the "all samples' link uses `...?sampleType=Serverless%20function` we don't need more here.

Reply:
Yes value is the value from the URL only, but typeLabel is the translated string in other samples. If we change the language, the type label will show in the respective language.

So if I look only for typeLabel, then if it is translated string, later we have to ask serverless team to add one metadata.label to indicate this is the sample wrt serverless functions

Now while creating the sample, if they add new label as sample-type : 'Serverless Function'  then if typeLabel is translated string also we will not be having any issue.